packages/contracts-bedrock/src/L2/TeaWAPOracle.sol

```diff
@@ -152,7 +274,15 @@ contract TeaWAPOracle {
         require(msg.sender == Ownable(Predeploys.PROXY_ADMIN).owner(), "TeaWAPOracle: admin only");
 
         require(_oracle != address(0), "TeaWAPOracle: zero address");
-        require(_twapObservations > 0, "TeaWAPOracle: zero observations");
+
+        // SECURITY FIX: Validate TWAP observations are within safe bounds
+        // REASON: Too few (1-2) = vulnerable to single-block manipulation
+        //         Too many (>100) = may fail due to insufficient historical data
+        require(
+            _twapObservations >= MIN_TWAP_OBSERVATIONS && _twapObservations <= MAX_TWAP_OBSERVATIONS,
+            "TeaWAPOracle: observations out of bounds"
+        );
+
```

Comment:
Not needed, see above (although I agree with your assessment and we plan to use 10).